src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,53 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_MIN_AGE = 18;
+    private static final int USER_PASSWORD_MIN_LENGTH = 6;
+    private static final int USER_LOGIN_MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUserInput(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUserInput(User user) {
+        validateUserExist(user);
+        validateUserLogin(user.getLogin());
+        validateUserPassword(user.getPassword());
+        validateUserAge(user.getAge());
```

Comment:
validate existence is an expensive operation, so lets' move it after the the simple ones
```suggestion
        validateUserLogin(user.getLogin());
        validateUserPassword(user.getPassword());
        validateUserAge(user.getAge());
        validateUserExist(user);
```